infrastructure/src/main/java/org/wso2/carbon/testgrid/infrastructure/aws/AWSManager.java

```diff
@@ -0,0 +1,198 @@
+                    "AWS", "Scripts", script.getFilePath())));
+            stackRequest.setTemplateBody(file);
+            log.info("Created a CloudFormation Stack with the name :" + stackRequest.getStackName());
+            stackbuilder.createStack(stackRequest);
+            waitForAWSProcess(stackbuilder, cloudFormationName);
+            DescribeStacksRequest describeStacksRequest = new DescribeStacksRequest();
+            describeStacksRequest.setStackName(cloudFormationName);
+            DescribeStacksResult describeStacksResult = stackbuilder.describeStacks(describeStacksRequest);
+            //Initially only PublicDNS which is exported from a dummy script is saved in the Deployment object.
+            List<Host> hosts = new ArrayList<>();
+            for (Stack st : describeStacksResult.getStacks()) {
+                for (Output output : st.getOutputs()) {
+                    if (output.getOutputKey().equals("PublicDNS")) {
+                        Host host = new Host();
+                        host.setIp(output.getOutputValue());
+                        hosts.add(host);
+                    }
+                }
+            }
+            Deployment deployment = new Deployment();
+            deployment.setHosts(hosts);
+            return deployment;
+        } catch (InterruptedException e) {
+            throw new TestGridInfrastructureException("Error occured while waiting for " +
+                    "CloudFormation Stack creation", e);
+        } catch (IOException e) {
+            throw new TestGridInfrastructureException("Error occured while Reading CloudFormation script", e);
+        }
+
+    }
+
+    /**
+     * This method waits for completion of AWS process and generate result depending on
+     * the result code.
+     *
+     * @param stackBuilder AWS CF builder object.
+     * @param stackName    The stack name to perform the waiting upon.
+     * @return true or false depending on the result.
+     * @throws InterruptedException            when an Interrupt occurs while waiting for CF result.
+     * @throws TestGridInfrastructureException when an error occurs while reading the cf template file.
+     */
+    private boolean waitForAWSProcess(AmazonCloudFormation stackBuilder, String stackName) throws InterruptedException,
+            TestGridInfrastructureException {
+        DescribeStacksRequest wait = new DescribeStacksRequest();
+        wait.setStackName(stackName);
+        boolean completed = false;
+        boolean successful = false;
+        log.info("Waiting ..");
+        while (!completed) {
+            List<Stack> stacks = stackBuilder.describeStacks(wait).getStacks();
+            if (stacks.isEmpty()) {
+                throw new TestGridInfrastructureException("There is no stack for the name :" + stackName);
+            } else {
+                for (Stack stack : stacks) {
+                    if (stack.getStackStatus().equals(StackStatus.CREATE_COMPLETE.toString()) ||
```

Comment:
Always compare the variable against the constant value to avoid possible null pointers.

Reply:
Changed the code